### PR TITLE
Create part1_exam

### PR DESCRIPTION
**Pull request by tinhdang**
Create questions in Part 1

### CODE REVIEW

File: app/Http/Controllers/Backend/ExamController.php [hunk 1]
```diff
@@ -113,4 +118,59 @@ public function destroy($id)
...
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request of exams
+     * @param int                      $id      of exam
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function storePart1(PostPart1Request $request, $id)
```
Comment (ghost) [inline app/Http/Controllers/Backend/ExamController.php]: I think you should move this function to QuestionController

File: app/Http/Controllers/Backend/ExamController.php [hunk 2]
```diff
@@ -113,4 +118,59 @@ public function destroy($id)
...
+    {
+        $requestQuestions = $request->all();
+        $questions = collect([]);
```
Comment (ghost) [inline app/Http/Controllers/Backend/ExamController.php]: this variables is not used

File: resources/views/backend/questions/create/part1.blade.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                <tr id="item">
+                  <td>{{$i}}</td>
+                  <td>
+                    <div class="form-group">
+                    <input type="text" name="question[{{$i}}][content][]" value="{{trans('questions.keyA')}}" readonly=""> 
+                    <input type="text" name="question[{{$i}}][content][]" value="{{trans('questions.keyB')}}" readonly=""> 
+                    <input type="text" name="question[{{$i}}][content][]" value="{{trans('questions.keyC')}}" readonly=""> 
+                    <input type="text" name="question[{{$i}}][content][]" value="{{trans('questions.keyD')}}" readonly=""> 
+                    <select name="question[{{$i}}][is_correct]"> 
+                      <option value="0">{{trans('questions.keyA')}}</option>
```
Comment (ghost) [inline resources/views/backend/questions/create/part1.blade.php]: pls define a const `NUMBER_OF_ANSWER`   =  4 in Question model, then in the view loop and render follow by that const

File: resources/views/backend/questions/create/part1.blade.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                <th style="width: 30%">{{trans('questions.picture')}}</th>
+              </tr>
+              </thead>
+              <tbody id ="add_question">
+              @for($i = 0; $i < \App\Models\Part::NUMBER_QUESTION_PART_1; $i++ )
+                <tr id="item">
+                  <td>{{$i}}</td>
+                  <td>
+                    <div class="form-group">
+                    <input type="text" name="question[{{$i}}][content][]" value="{{trans('questions.keyA')}}" readonly="">
```
Comment (ghost) [inline resources/views/backend/questions/create/part1.blade.php]: Define the const number of answer and don't need these input

File: app/Http/Controllers/Backend/ExamController.php [hunk 5]
```diff
@@ -113,4 +118,59 @@ public function destroy($id)
...
+        DB::transaction(function () use ($requestQuestions, $id, $questions) {
+            for ($i = 0; $i < count($requestQuestions['question']); $i++) {
+                $question = new Question;
+                $question->exam_id = $id;
+                $question->part_id = \App\Models\Part::PART_1;
+                $question->save();
+
+                $questionImage = new QuestionImage;
+                $requestQuestionsI = $requestQuestions['question'][$i];
```
Comment (ghost) [inline app/Http/Controllers/Backend/ExamController.php]: why you name it `$requestQuestionsI`, this name is not good

File: app/Models/Exam.php [hunk 6]
```diff
@@ -12,6 +12,11 @@ class Exam extends Model
     const FINISHED = 1;
 
     /**
+     * Flag type finish_ing exams
+     */
+    const FINISH_ING = 2;
```
Comment (ghost) [inline app/Models/Exam.php:17]: What does it mean?

**Review by ghost — CHANGES_REQUESTED** [review]
pls check my cmt

File: app/Models/Exam.php (hunk 6 above)
Comment (tinhdang) [reply app/Models/Exam.php:17]: It mean when exam don't finished , can continue  edit & add this exam

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**